Receiver.cs

```diff
@@ -18,6 +16,16 @@ static internal void Run(UdtSocket conn)
             using (var writer = new BinaryWriter(netStream))
             using (var reader = new BinaryReader(netStream))
             {
+                // transmit your role and check if connected peer has the opposite role
+                writer.Write(myRole.ToString());
+                string role = reader.ReadString();
+
+                if (role == myRole.ToString())
+                {
+                    Console.Error.WriteLine("Peers can't have the same role.");
+                    return;

```

Comment:
if you exit the function like this, this won't translate into a non-zero exitcode for the program

Reply:
What do you suggest instead?

Reply:
call directly Environment.Exit? I know it doesn't seem very clean, but it would be more correct.